include/xtensor/xcontainer.hpp

```diff
@@ -368,6 +252,84 @@ namespace xt
 
 #undef DL
 
+    template <class D>
```

Comment:
These methods should be defined with after the other ones from `xcontainer`, just before the definition of `derived_cast`.